addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/AutoDetectAuthenticationMethodType.java

```diff
@@ -222,7 +222,7 @@ public AuthMethodApiResponseRepresentation(Map<String, T> values) {
         @Override
         public JSON toJSON() {
             JSONObject response = new JSONObject();
-            response.put(getName(), super.toJSON());
+            response.put(super.getName(), super.toJSON());
```

Comment:
Why is this being changed?

Reply:
I'll have to check the rule/finding

Reply:
I couldn't find a link to the rule, but here's the scoop:

![image](https://github.com/user-attachments/assets/f90803e1-bd79-4f3b-8b3b-87f97c728843)

Reply:
How is it going to access "`Outer.this`" if the class is static?

Reply:
I guess the rule doesn't account for that. I can revert.

Reply:
Would be better.